src/ripple/protocol/impl/STParsedJSON.cpp

```diff
@@ -542,18 +542,20 @@ static boost::optional<detail::STVar> parseLeaf (
 
                         std::string const strValue (account.asString ());
 
-                        if (value.size () == 40) // 160-bit hex account value
-                            uAccount.SetHex (strValue);
+                        // If we have what looks like a 160-bit hex account
+                        // value, try it and see if it's valid.
+                        bool const success = (value.size () == 40)
+                            ? uAccount.SetHex (strValue)
+                            : false;
 
+                        if (!success)
```

Comment:
You never use this variable again - so why even name it?

   if (! (value.size () == 40 && uAccount.SetHex (strValue))